edx_solutions_api_integration/users/views.py

```diff
@@ -1554,3 +1560,31 @@ def post(self, request, user_id, msg_id):
         mark_notification_read(int(user_id), int(msg_id), read=bool(read))
 
         return Response({}, status=status.HTTP_201_CREATED)
+
+
+class UsersCourseProgressList(SecureListAPIView):
+    """
+    The UsersCourseProgressList view allows you to retrieve a list of courses user enrolled in and the progress
+    for a user
+    - URI: ```/api/users/{user_id}/courses/progress```
+    """
+    pagination_class = None
+
+    def get(self, request, user_id):  # pylint: disable=unused-argument
+        try:
+            user = User.objects.get(id=user_id)
+        except ObjectDoesNotExist:
+            raise Http404
+        queryset = CourseEnrollment.objects.filter(user=user)
+
+        student_progress = StudentProgress.objects.filter(user=user).values('course_id', 'completions')
+        course_overview = CourseOverview.objects.all().values('id', 'start', 'end', 'course_image_url', 'display_name')
```

Comment:
I have concerns with pulling everything from `CourseOverview` and `CourseAggregatedMetaData` as iterating over large dataset would have performance impact.